source/http/httpsource.go

```diff
@@ -172,6 +181,9 @@ func (hs *httpSourceHandler) CacheKey(ctx context.Context, index int) (string, b
 		if err == nil {
 			if resp.StatusCode == http.StatusOK || resp.StatusCode == http.StatusNotModified {
 				respETag := resp.Header.Get("ETag")
+				if respETag == "" && onlyETag != "" {
```

Comment:
shouldn't this only run on `resp.StatusCode == http.StatusNotModified` ?

Reply:
Hmm. Yes, you're right. This would cause an incorrect ETag to be associated in the following scenario:
  * The remote server had in the past responded with an ETag for this resource, which was cached.
    - Otherwise, `onlyETag` would be empty.
  * That was the _only_ ETag cached for this resource.
    - Otherwise, `onlyETag` would be empty.
  * The remote server then *stopped* supporting `ETag`/`If-None-Match` for this resource.
    - Otherwise, it would respond with a 304 or a 200+ETag.

I'll update. Thank you for catching the error.

Reply:
Fixed.